Game/RoomsAndDoors/RoomFactory.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Xml.Linq;
+using MainGame.Doors;
+using MainGame.Rooms;
+using MainGame.SpriteHandlers;
+using Microsoft.Xna.Framework;
+using MainGame.Blocks;
+using MainGame.Items;
+using MainGame.Managers;
+
+namespace MainGame.RoomsAndDoors
+{
+    public static class RoomFactory
+    {
+        /*
+         * Method for generating a room object based on a room name as specified in Content/Rooms
+         */
+        public static Room GenerateRoom(string roomName, Game1 game)
+        {
+            Room room = null;
+
+            string path = "Content\\Rooms\\" + roomName + ".csv";
```

Comment:
So this file path finding method only works on Windows since UNIX based systems uses a forward slash instead of backslash (see [here](https://stackoverflow.com/questions/38428561/difference-between-forward-slash-and-backslash-in-file-path)). It think it's best if you used the Path.Combin() method from System.IO. Like this, `Path.Combine("Content", "Rooms", $"{roomName}.csv")`